From 7f6d52ac7bdf91d9e1d0750ae3afd0d128c8ac94 Mon Sep 17 00:00:00 2001
From: Timo Gurr <tgurr@exherbo.org>
Date: Thu, 25 Oct 2012 02:12:29 +0200
Subject: [PATCH] Disable tests that fail with sydbox

Edit: Daniel Mierswa <impulze@impulze.org>
I removed the disabling of 2 tests here since they seem to work fine
with 7.5. There's another test failing on my machine (tail-2/assert)
but since it works for alip, let's see if more people run into it.

Edit: Elias Pipping <pipping@exherbo.org>
All but two tests that originally failed still do (the others are
skipped automatically if strace does not work). Made the patch apply
again

Note by Paul Seidler <sepek@exherbo.org>:
tests/install/basic-1.sh and tests/mkdir/p-3.sh failed for me even without sydbox
---
 tests/install/basic-1.sh     | 2 ++
 tests/mkdir/p-3.sh           | 2 ++
 tests/tail-2/inotify-race.sh | 2 ++
 3 files changed, 6 insertions(+)

diff --git a/tests/install/basic-1.sh b/tests/install/basic-1.sh
index 08055f5..c6f9805 100755
--- a/tests/install/basic-1.sh
+++ b/tests/install/basic-1.sh
@@ -92,6 +92,8 @@ test -d xx/rel && fail=1
 # inaccessible parent.  coreutils 5.97 fails this test.
 # Perform this test only if "." is on a local file system.
 # Otherwise, it would fail e.g., on an NFS-mounted file system.
+
+skip_ "Test fails under sydbox, see bug #204. -alip"
 if is_local_dir_ .; then
   mkdir -p sub1/d || fail=1
   (cd sub1/d && chmod a-r . && chmod a-rx .. &&
diff --git a/tests/mkdir/p-3.sh b/tests/mkdir/p-3.sh
index a80fa37..a50757b 100755
--- a/tests/mkdir/p-3.sh
+++ b/tests/mkdir/p-3.sh
@@ -38,6 +38,8 @@ test -d "$p/b/b" && test -d "$p/z" || fail=1
 # whose parent is inaccessible.  coreutils 5.97 fails this test.
 # Perform this test only if "." is on a local file system.
 # Otherwise, it would fail e.g., on an NFS-mounted file system.
+skip_ "Test fails under sydbox, see bug #204. -alip"
+
 if is_local_dir_ .; then
   (cd no-acce3s/d && chmod a-r . && chmod a-rx .. &&
       mkdir -p a/b "$p/b/c" d/e && test -d a/b && test -d d/e) || fail=1
diff --git a/tests/tail-2/inotify-race.sh b/tests/tail-2/inotify-race.sh
index 7476283..3d94358 100755
--- a/tests/tail-2/inotify-race.sh
+++ b/tests/tail-2/inotify-race.sh
@@ -23,6 +23,8 @@
 . "${srcdir=.}/tests/init.sh"; path_prepend_ ./src
 print_ver_ tail
 
+skip_ 'This test (using gdb) hangs under sydbox'
+
 # Don't run this test by default because sometimes it's skipped as noted below.
 # Also gdb has a bug in Debian's gdb-6.8-3 at least that causes it to not
 # cleanup and exit correctly when it receives a SIGTERM, thus hanging the test.
-- 
1.7.12.4

